src/interface.jl

```diff
@@ -209,4 +233,19 @@ function evaluate_ed2!(flex_B::FlexArray,
    B, dB, ddB = _alloc_ed2(flex_B, flex_dB, flex_ddB, basis, x)
    evaluate_ed2!(B, dB, ddB, basis, x)
    return B, dB, ddB
+end
+
+# --------------------------------------- 
+# general rrule and frule interface for ChainRulesCore
```

Comment:
Is it deliberate that the rrule here is defined only for real inputs? Complex is a little trickier but can be done. In a way I don't mind if we first take care of real only but then please open an issue to document that the complex rrules are still missing.

Reply:
Since what we used before was just real inputs I did that restriction. I will open an issue to note this is not implemented yet.

Reply:
I think this issue is now open